### PR TITLE
remove tools/setup_helpers/cudnn.py

### PR DESCRIPTION
**Pull request by xuhdev**
FindCUDNN.cmake and cuda.cmake have done the detection. This commit deletes `tools/setup_helpers/cudnn.py` as it is no longer needed.

### CODE REVIEW

File: tools/setup_helpers/cudnn.py [hunk 1]
```diff
@@ -1,97 +0,0 @@
...
-        '/usr/include/',
-    ] + gather_paths([
-        'CPATH',
-        'C_INCLUDE_PATH',
-        'CPLUS_INCLUDE_PATH',
-    ])))
-    # Add CUDA related dirs to candidate list
-    if IS_CONDA:
-        lib_paths.append(os.path.join(CONDA_DIR, 'lib'))
-        include_paths.append(os.path.join(CONDA_DIR, 'include'))
```
Comment (xuhdev) [inline tools/setup_helpers/cudnn.py]: An equivalent to these two lines are at https://github.com/pytorch/pytorch/blob/0b1fee0819c82736cef387c3f02b7aa4c75419aa/tools/setup_helpers/cmake.py#L210

File: tools/setup_helpers/cudnn.py [hunk 2]
```diff
@@ -1,97 +0,0 @@
...
-WITH_STATIC_CUDNN = os.getenv("USE_STATIC_CUDNN")
-
-if USE_CUDA and not check_negative_env_flag('USE_CUDNN'):
-    lib_paths = list(filter(bool, [
-        os.getenv('CUDNN_LIB_DIR')
-    ] + lib_paths_from_base(CUDA_HOME) + [
-        '/usr/lib/x86_64-linux-gnu/',
-        '/usr/lib/powerpc64le-linux-gnu/',
-        '/usr/lib/aarch64-linux-gnu/',
-        '/usr/lib/',
```
Comment (xuhdev) [inline tools/setup_helpers/cudnn.py]: CMake is able to detect architecture and therefore finds the correct dir without the need to specifying them explicitly

**Comment by xuhdev** [comment]
Hope I've got as many CIs up as possible...

**Comment by xuhdev** [comment]
@ezyang These CI failures are probably bogus, as #25718 also shows that merely a CI change causes the same failure.

Feel free to revert 83e97941fcd503af6c7358df8ee19a4841001340 (or ask me to) if you think it's ready to merge

**Review by ezyang — APPROVED** [review]
Relevant job appears to have succeeded: https://circleci.com/gh/pytorch/pytorch/2685308?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by ezyang** [comment]
Noooo it failed again

```
Sep 06 14:39:42 CMake Error at cmake/public/utils.cmake:20 (get_target_property):
Sep 06 14:39:42   get_target_property() called with non-existent target
Sep 06 14:39:42   "nvonnxparser_static".
Sep 06 14:39:42 Call Stack (most recent call first):
Sep 06 14:39:42   cmake/Dependencies.cmake:1088 (caffe2_interface_library)
Sep 06 14:39:42   CMakeLists.txt:365 (include)
```

This time in caffe2_py2_cuda9_0_cudnn7_ubuntu16_04_build

**Comment by xuhdev** [comment]
:( OOOO... I'll look into it.

**Comment by xuhdev** [comment]
Strangely, caffe2_py2_cuda9_1_cudnn7_ubuntu16_04_build succeeded:

https://circleci.com/gh/pytorch/pytorch/2695311?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

**Comment by xuhdev** [comment]
Feel free to revert for now -- Likely I have to tweak with CI and it will take a while

**Comment by xuhdev** [comment]
@ezyang It looks like to enable more CIs all at once I have to push to a new branch named `ci-all/*` **in this repo**. Do you know whether this would cause any undesirable effects? I just wanna make sure there would be no unexpected deployment and such (I didn't see any `ci-all/*` branch in this repo, which makes me doubt a bit).

**Comment by ezyang** [comment]
@xuhdev Nope, you can do that :)

**Pull request closed without merging**